tests/arrays.test.ts

```diff
@@ -0,0 +1,198 @@
+import { createTestSchema, runPlugin } from "./utils/test-helpers";
+
+describe("Array Handling", () => {
+  const schemaWithArrays = `
+    type Author {
+      id: ID!
+      name: String!
+    }
+
+    type Book {
+      id: ID!
+      title: String!
+      authors: [Author!]!
+      coAuthors: [Author!]
+      tags: [String!]!
+      optionalTags: [String!]
+      categories: [String]!
+      optionalCategories: [String]
+    }
+
+    # DTO type to test mutable arrays
```

Comment:
nit: i'd prefer "Object" over "DTO".  This is how they are referred to in the [spec](https://spec.graphql.org/draft/#sec-Objects).

Reply:
Right, I'm using DTO to match the parlance used throughout the historical comments and git commit history.